packages/phone-number-privacy/combiner/src/bls/bls-cryptography-client.ts

```diff
@@ -1,57 +1,44 @@
-import { AzureKeyVaultClient } from '@celo/contractkit/lib/utils/azure-key-vault-client'
-import threshold from 'blind-threshold-bls'
-import { ErrorMessages } from '../common/error-utils'
+import threshold_bls from 'blind-threshold-bls'
 import logger from '../common/logger'
-import config, { DEV_MODE, DEV_PRIVATE_KEY } from '../config'
+import config from '../config'
 
+export interface ServicePartialSignature {
+  url: string
+  signature: string
+}
+
+function flattenSigsArray(sigs: Uint8Array[]) {
+  return Uint8Array.from(sigs.reduce((a, b) => Array.from(a).concat(Array.from(b)), [] as any))
+}
 export class BLSCryptographyClient {
   /*
    * Computes the BLS signature for the blinded phone number.
    */
-  public static async computeBlindedSignature(base64BlindedMessage: string) {
-    try {
-      const privateKey = await BLSCryptographyClient.getPrivateKey()
-      const keyBuffer = Buffer.from(privateKey, 'base64')
-      const msgBuffer = Buffer.from(base64BlindedMessage, 'base64')
-
-      logger.debug('Calling theshold sign')
-      const signedMsg = threshold.signBlindedMessage(keyBuffer, msgBuffer)
-      logger.debug('Back from threshold sign, parsing results')
-
-      if (!signedMsg) {
-        throw new Error('Empty threshold sign result')
+  public static async combinePartialBlindedSignatures(
+    serviceResponses: ServicePartialSignature[],
+    blindedMessage: string
+  ): Promise<string> {
+    const polynomial = config.thresholdSignature.polynomial
+    const sigs: Uint8Array[] = []
+    for (const service of serviceResponses) {
+      const sigBuffer = Buffer.from(service.signature, 'base64')
+      try {
+        await threshold_bls.partialVerifyBlindSignature(
+          Buffer.from(polynomial, 'base64'),
+          Buffer.from(blindedMessage, 'base64'),
+          sigBuffer
+        )
+        sigs.push(sigBuffer)
+      } catch (e) {
+        logger.warn(`could not verify signature for service url ${service.url}`)
       }
-
-      return Buffer.from(signedMsg).toString('base64')
-    } catch (e) {
-      logger.error(ErrorMessages.SIGNATURE_COMPUTATION_FAILURE, e)
-      throw e
-    }
-  }
-
-  private static privateKey: string
-
-  /**
-   * Get singleton privateKey
-   */
-  private static async getPrivateKey(): Promise<string> {
-    if (DEV_MODE) {
-      return DEV_PRIVATE_KEY
     }
-
-    if (BLSCryptographyClient.privateKey) {
-      return BLSCryptographyClient.privateKey
+    const threshold = config.thresholdSignature.threshold
+    if (sigs.length < threshold) {
+      logger.error(`not enough not enough partial signatures'${sigs.length}'/'${threshold}'`)
```

Comment:
nit: "not enough" dupes

Reply:
Also, I don't think the single quote chars are necessary since you're using the `

Reply:
ya -- copy pasta.  i'll fix it up